include/motor-control/core/stepper_motor/motion_controller.hpp

```diff
@@ -110,3 +109,96 @@ class MotionController {
 };
 
 }  // namespace motion_controller
+
+namespace pipette_motion_controller {
+
+using namespace motor_messages;
+using namespace motor_hardware;
+
+// TODO(lc 05-22-2022) We should have PipetteMotionController inherit
+// from MotionController, but that will require some template refactoring
+// to make it compile.
+template <lms::MotorMechanicalConfig MEConfig>
+class PipetteMotionController {
+  public:
+    using GenericQueue = freertos_message_queue::FreeRTOSMessageQueue<Move>;
+    PipetteMotionController(lms::LinearMotionSystemConfig<MEConfig> lms_config,
+                            PipetteStepperMotorHardwareIface& hardware_iface,
+                            MotionConstraints constraints, GenericQueue& queue)
+        : linear_motion_sys_config(lms_config),
+          hardware(hardware_iface),
+          motion_constraints(constraints),
+          queue(queue),
+          steps_per_mm(convert_to_fixed_point_64_bit(
+              linear_motion_sys_config.get_steps_per_mm(), 31)) {}
+
+    auto operator=(const PipetteMotionController&)
+        -> PipetteMotionController& = delete;
+    auto operator=(PipetteMotionController&&)
+        -> PipetteMotionController&& = delete;
+    PipetteMotionController(PipetteMotionController&) = delete;
+    PipetteMotionController(PipetteMotionController&&) = delete;
+
+    ~PipetteMotionController() = default;
+
+    [[nodiscard]] auto get_mechanical_config() const
+        -> const lms::LinearMotionSystemConfig<MEConfig>& {
+        return linear_motion_sys_config;
+    }
+
+    void move(const can_messages::TipActionRequest& can_msg) {
+        steps_per_tick velocity_steps =
+            fixed_point_multiply(steps_per_mm, can_msg.velocity);
+        auto stop_condition = MoveStopCondition::none;
+        if (check_tip_sense()) {
```

Comment:
we definitely cannot do this, at least long term - what if something gets stuck on and pushing again might pop it off? We need to have a way for higher levels with more context to command tip actions that don't necessarily make sense to this level of the code. the tip action message should specify what kind of action it needs to take.